packages/playground/remote/src/lib/worker-thread.ts

```diff
@@ -188,6 +189,59 @@ export class PlaygroundWorkerEndpoint extends PHPWorker {
 	}
 }
 
+async function downloadWordPressAssets(php: PHP) {
+	if (!php.requestHandler) {
+		logger.warn('No PHP request handler available');
+		return;
+	}
+	const wpVersion = await getLoadedWordPressVersion(php.requestHandler);
+	const staticAssetsDirectory = wpVersionToStaticAssetsDirectory(wpVersion);
+	if (!staticAssetsDirectory) {
+		logger.warn('No static assets directory for WordPress', wpVersion);
+		return;
+	}
+	const response = await fetch(
+		[wordPressSiteUrl, staticAssetsDirectory, 'wordpress-static.zip'].join(
+			'/'
+		)
+	);
+
+	if (!response.ok) {
+		logger.warn('Failed to download WordPress assets');
+		return;
+	}
+	if (!response.body) {
+		logger.warn('Response body is not readable');
+		return;
+	}
+	try {
+		const zipStream = decodeZip(response.body);
```

Comment:
Unfortunately this is much slower than downloading the entire zip file, passing it to PHP, and uncompressing it there, see https://github.com/WordPress/wordpress-playground/pull/851#issuecomment-1892010942.

Once we have a better stream interop, we could pass the stream object directly where PHP would start unzipping it early – this should be the fastest method of all.

Reply:
On a second thought, maybe that's fine here as:

* We're not blocking the interactions 
* We're running this directly in the worker without passing messages around

And I like stream-processing in that it would retain some of these files in case of a mid-download page refresh.